src/app/EventManagement.cpp

```diff
@@ -33,6 +33,21 @@ namespace chip {
 namespace app {
 static EventManagement sInstance;
 
+#if CHIP_SYSTEM_CONFIG_NO_LOCKING
+static void LockEvents() {}
+static void UnLockEvents() {}
+#else
+static System::Mutex sEventLock;
+static void LockEvents()
```

Comment:
EventManagement is already a global singleton, so it'd be cleaner to add LockEvents() as a public API method on that object.

Moreover, if we plan to use any #ifdef treatment here, the actual ifdef should be moved within the LockEvent() implementation itself, not outside it.

Reply:
If you move it in, then the "Events" part of the "LockEvents" is kind of redundant.

Reply:
Done, thanks